contracts/src/systems/actions.cairo

```diff
@@ -100,21 +100,45 @@ mod actions {
-            // Get seeds from the selected pit and validate it's not empty
-            let mut seeds = mancala_game.get_seeds(current_player, selected_pit);
-            if seeds == 0 {
-                panic!("Selected pit is empty. Choose another pit.");
-            }
-            mancala_game.clear_pit(ref current_player, selected_pit);
-            mancala_game
-                .distribute_seeds(ref current_player, ref opponent, ref seeds, selected_pit);
-            if mancala_game.is_game_finished(current_player, opponent) {
-                mancala_game.set_winner(current_player, opponent);
+
+            if mancala_game.status == GameStatus::InProgress {
+                let (mut current_player, mut opponent) = mancala_game.get_players(world);
+                // Get seeds from the selected pit and validate it's not empty
+                let mut seeds = mancala_game.get_seeds(current_player, selected_pit);
+                if seeds == 0 {
+                    panic!("Selected pit is empty. Choose another pit.");
+                }
+                mancala_game.clear_pit(ref current_player, selected_pit);
+                mancala_game
+                    .distribute_seeds(ref current_player, ref opponent, ref seeds, selected_pit);
+                if mancala_game.is_game_finished(current_player, opponent) {
+                    mancala_game.status = GameStatus::Finished;
+                    mancala_game.set_winner(current_player, opponent);
+                }
+                set!(world, (mancala_game, current_player, opponent));
+                // return the current player so client has ability to know
+                (mancala_game.current_player, mancala_game.status)
+            } else {
+                set!(world, (mancala_game));
+                (mancala_game.current_player, mancala_game.status)
             }
-            set!(world, (mancala_game, current_player, opponent));
-            // return the current player so client has ability to know
-            (mancala_game.current_player, mancala_game.status)
+        }
+
+        // set the game as `TimeOut` if a player hasn't made a move
+        fn time_out(world: IWorldDispatcher, game_id: u128) {
+            let mut mancala_game: MancalaGame = get!(world, game_id, (MancalaGame));
+            assert!(mancala_game.status == GameStatus::InProgress, "Game is not in progress");
+
+            let (_, mut opponent) = mancala_game.get_players(world);
+            let execution_info = get_execution_info_syscall().unwrap_syscall().unbox();
+            let block_info = execution_info.block_info.unbox();
+            assert!(
+                block_info.block_number >= mancala_game.last_move + mancala_game.time_between_move,
+                "Game is in progress"
+            );
+
+            mancala_game.status = GameStatus::TimeOut;
+            mancala_game.winner = opponent.address;
         }
```

Comment:
Code looks awesome but we need the set!() call at the end of this function so the data is written on chain

Reply:
yes true, i forgot it thanks!